userspace/libscap/engine/bpf/bpf_public.c

```diff
@@ -1,24 +0,0 @@
-/*
-Copyright (C) 2022 The Falco Authors.
-
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-
-    http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
-
-#include "bpf_public.h"
-#include <stdlib.h>
-
-const char *scap_get_bpf_probe_from_env()
-{
-	return getenv(SCAP_BPF_PROBE_ENV_VAR_NAME);
-}
-
```

Comment:
I don't know why we use this approach to open the bpf probe, but I had to change it since in this way we discover only during the scap_open phase if we are using the bpf engine or not and this is really strange in my opinion, and btw this method cannot work with the new approach proposed int his PR where the client must know which engine it will use! I think this is the behavior we want but I'm all ears if you other suggestions